dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+#include "pch.h"
+
+const std::wstring ConvertByteArrayToWideString(unsigned int payloadLength, _In_reads_(payloadLength) byte* payload);
+void ProtocolLaunchHelper(unsigned int payloadLength, _In_reads_(payloadLength) byte* payload);
+
+inline HRESULT GetAppUserModelId(wil::unique_cotaskmem_string& appUserModelId)
+{
+    wchar_t appId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+    UINT32 appIdSize{ ARRAYSIZE(appId) };
+
+    GetCurrentApplicationUserModelId(&appIdSize, appId);
+
+    appUserModelId = wil::make_unique_string<wil::unique_cotaskmem_string>(appId);
```

Comment:
I believe this throws for the no memory case. But your helper method doesn't handle that.